### ISSUE TITLE
add connect_with_timeout to set connection timeout

### ISSUE THREAD

**Issue by XuGuangnian** [1148118271/ssh-rs#90]
I need set the connection timeout with shorter Duration and R/W timeout with longer Duration, so add new method  to set connection timeout instead of the global timeout.

```rust
fn main() {
    let mut session = ssh::create_session()
        .username("root")
        .password("root")
        .timeout(Some(Duration::from_millis(5000)))
        // .connect("192.168.220.134:22")
        .connect_with_timeout("192.168.220.134:22", Some(Duration::from_millis(1000)))
        .unwrap()
        .run_local();
    // Usage 1
    let mut shell = session.open_shell().unwrap();
    run_shell(&mut shell);

    // Close channel.
    shell.close().unwrap();
    // Close session.
    session.close();
}
```

**Comment by HsuJv** [comment]
Hi there,

Your idea looks good, but actually, there already is an API `connect_bio` that provides the feasibility of customising the underlying connection in any way.

I'm wondering if it is necessary to add more APIs to customise only TCP streams.

Is there any user story that `connect_bio` cannot satisfy?

BRs.

**Comment by XuGuangnian** [comment]
Thank you for you reply.

I think it would be better to add `connect_with_timeout` method. 

1. There is a curving way to add a connection timeout by customising the TcpStream, including error handling (cannot reuse the error handling with `SshError`).
2. The connection timeout config is a very basic and commonly used feature for networking crates.
3. One parameter configuration `timeout` indicates two meanings, which may lead to misunderstanding. (yes, that's me)

Also, a alternative way is to add a timeout param for `connect` API, but it's not appropriate for minor version updates.

**Comment by HsuJv** [comment]
fair enough,
I can do the update later this week

BRs.

**Comment by HsuJv** [comment]
Hi, sorry for the late,

I've released a new version of `ssh-rs` 0.5.0, which includes your change.
Unfortunately, during the fix of another issue API change was necessary, so the minor version also increased.

BRs.
